src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidRunCommand.cs

```diff
@@ -205,6 +205,7 @@ protected override Task<ExitCode> InvokeInternal(ILogger logger)
                 if (instrumentationExitCode != expectedExitCode)
                 {
                     logger.LogError($"Non-success instrumentation exit code: {instrumentationExitCode}, expected: {expectedExitCode}");
+                    return ExitCode.TESTS_FAILED;
```

Comment:
FWIW, when expected doesn't match actual exit code, we return `ExitCode.GENERAL_FAILURE` in WASM/iOS

Reply:
how do you differ failed tests from `GENERAL_FAILURE` in that case? When expected doesn't match actual exit code isn't it a failed test?

Reply:
It's a bit different in iOS/WASM - for `test` command we don't try to detect exit code at all and only decide based on how the tests went. For `run` command, we don't expect tests inside and try to detect the exit code and compare it with expected.

So it's a little bit different situation and might not apply here. I only meant that as "let's behave similar in Android/iOS" but if that doesn't make sense, it's not a big deal.